### PR TITLE
policy: ssh check mode

### LINKED ISSUE #1850: Support for SSH check mode in ACLs
## Why

Tailscale upstream supports SSH check mode.

We would like to use it with Headscale, too.

## Description

When defining an Tailscale SSH ACL policy with the `action` set to `check`, an additional authentication against the OIDC endpoint is required, which grants access within an optional `checkPeriod`, defaulting to 12 hours and allowing to be set to `always`.

## References

* [Configure Tailscale SSH with check · Tailscale SSH · Tailscale Docs](https://tailscale.com/kb/1193/tailscale-ssh#configure-tailscale-ssh-with-check-mode)

This is related to, but not identical to:

- #1303, where a user tried to use `check` mode
- #1623, where upstream reports about the availability of a web check mode in the Tailscale client, currently restricted to the upstream control plane.

### PR DESCRIPTION
**Pull request by kradalby**
- Implement SSH check action (checkPeriod/checkLocal) — policy-driven SSH authorization via Noise protocol handler with
 server URL injection
- Add headscale auth register/approve CLI commands — new gRPC RPCs for programmatic node registration and approval
- Rewrite CLI internals — switch all commands to RunE, add grpcRun wrapper, deduplicate helpers, remove legacy namespace/machine aliases
- Replace gorilla/mux with chi and generalise auth templates/flow for the new /auth endpoint
- Add integration tests for SSH check modes (CLI + OIDC variants, negative cases, check-period) and switch tests to headscale auth register

Closes #1850

Claude was used, particularly in tests.

### CODE REVIEW

**Comment by nblock** [comment]
[Tailscale docs for `checkPeriod`](https://tailscale.com/docs/reference/syntax/policy-file#checkperiod):

- no `checkPeriod` provided: default to 12h
- `checkPeriod: always`: enforce a check with each request
- min value: 1m
- max: value:  168h (1 week)

---

- wire the chi logger into zerolog

---

Should there be a `headscale auth reject` subcommand to reject an auth request? The client will otherwise continuously query the server and create log entries until stopped ctrl-c by the user.

- "the code is there, so for web we could def give a reject and post a message"
- is it possible to cancel when OIDC is in use?

**Comment by nblock** [comment]
We'll need to check the evaluation order of SSH rules. From the relevant section in the [Tailscale docs (also contains examples)](https://tailscale.com/docs/features/tailscale-ssh#order-of-evaluation):

> SSH access rules are evaluated considering the most restrictive policies first:
> 
>  - Check policies
>  - Accept policies

**Comment by kradalby** [comment]
@nblock I think everything is addressed now. PTAL.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,19 @@ to understand how the packet filter should be generated. We discovered a few dif
 overall our implementation was very close.
 [#3036](https://github.com/juanfont/headscale/pull/3036)
 
+### SSH check action
+
+SSH rules with `"action": "check"` are now supported. When a client initiates an SSH connection to a node
```
Comment (nblock) [inline CHANGELOG.md]:
```suggestion
SSH rules with `"action": "check"` are now supported. When a client initiates a SSH connection to a node
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,19 @@ to understand how the packet filter should be generated. We discovered a few dif
...
+with a `check` action policy, the user is prompted to authenticate via OIDC or CLI approval before access
+is granted.
+
+A new `headscale auth` CLI command group supports the approval flow:
+
+- `headscale auth approve --auth-id <id>` approves a pending authentication request (SSH check or web auth)
+- `headscale auth register --auth-id <id> --user <user>` registers a node (replaces deprecated `headscale nodes register`)
```
Comment (nblock) [inline CHANGELOG.md:24]: Could you also list `headscale auth reject` ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -35,6 +50,11 @@ overall our implementation was very close.
 - **ACL Policy**: Merge filter rules with identical SrcIPs and IPProto matching Tailscale behavior - multiple ACL rules with the same source now produce a single FilterRule with combined DstPorts [#3036](https://github.com/juanfont/headscale/pull/3036)
 - Remove deprecated `--namespace` flag from `nodes list`, `nodes register`, and `debug create-node` commands (use `--user` instead) [#3093](https://github.com/juanfont/headscale/pull/3093)
 - Remove deprecated `namespace`/`ns` command aliases for `users` and `machine`/`machines` aliases for `nodes` [#3093](https://github.com/juanfont/headscale/pull/3093)
+- Add SSH `check` action support with OIDC and CLI-based approval flows [#1850](https://github.com/juanfont/headscale/pull/1850)
+- Add `headscale auth register` and `headscale auth approve` CLI commands [#1850](https://github.com/juanfont/headscale/pull/1850)
```
Comment (nblock) [inline CHANGELOG.md]: Probably also list `headscale auth reject` here.
